src/coreclr/minipal/Unix/CMakeLists.txt

```diff
@@ -0,0 +1,4 @@
+add_library(coreclrminipal
+    STATIC
+    doublemapping.cpp
```

Comment:
What you have here is fine, for now at least. I have been wondering whether we should build minipals like this in C to make easy to share with Mono when we need to, similar to what we are doing in libraries shims.

Reply:
Agreed, the physical location of minipal could be `src/native/minipal` (we can also move `src/native/common/getexepath.h` there). Maybe it's easier to do it now since there is only doublemapping.cpp file and it's basically C compatible code (except for the scope resolution operator `::`). :)

Reply:
I see the possibility of using this minipal by Mono as very remote. My vision is that it will gradually become a tiny replacement for the coreclr PAL, eventually allowing us to remove all Windowsisms from the coreclr. So I see this as a kind of inseparable part of the coreclr itself. While this double mapping stuff looks like something that Mono might possibly want, the other stuff that will go into the minipal is most likely going to be very coreclr specific. So I'd prefer keeping it in the coreclr folder until there is a reason to share it. 
As for converting it to C, we would need to prevent inlining of all the methods to be safe. Inlining C code into C++ is potentially dangerous e.g. due to c++ pointer aliasing. See https://github.com/dotnet/coreclr/pull/6801 for some details. We have converted all coreclr PAL files to c++ to avoid this issue in the past.

Reply:
Actually, we could still compile the minipal as C++ when building it for coreclr even if it was written in C, to avoid any C/C++ combination issues. So please disregard my comment on the inlining.

Reply:
Yes, it is fine to delay this until it is actually needed. I have mentioned this since I am regularly being asked by our leadership about what we can do share more code between CoreCLR and Mono. PAL layers are one of the easier opportunities.